ideas/2024.md

```diff
@@ -4,6 +4,22 @@ Welcome to our Google Summer of Code ideas list!
 
 ## small (90 hours)
 
+### Nix cleanup: merge `ParsedDerivation` into `BasicDerivation`, improve JSON format
+
+Nix currently special-cases some environment variables in derivations to convey extra meaning ("system features", "prefer local build", etc.).
+We're stuck with this in the [`.drv` A-Term format](https://nixos.org/manual/nix/stable/protocols/derivation-aterm) and `builtins.derivation` because backwards compatibility, but that doesn't mean our internal data structures and (currently experimental, i.e. not yet stabilized and immobile) JSON format cannot be improved.
```

Comment:
```suggestion
We're stuck with this in the [`.drv` A-Term format](https://nixos.org/manual/nix/stable/protocols/derivation-aterm) and `builtins.derivation` because of backwards compatibility requirements, but that doesn't mean our internal data structures and (currently experimental, i.e. not yet stabilized and immobile) JSON format cannot be improved.
```